### PR TITLE
 Add API  endpoints

### PR DESCRIPTION
**Pull request by ridabensalem**
in this milestone of the blog app project, we have added the following requirements : 
- Create an API endpoint to list all posts for a user.
- Create an API endpoint to list all comments for a user's post.
- Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should not be set by passing a parameter.
- Your API endpoints should receive JSON and respond JSON as well.

### CODE REVIEW

File: app/controllers/api/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class Api::V1::CommentsController < Api::V1::ApplicationController
+  def index
+    @post = Post.find(params[:post_id])
+
+    render json: @post.comments
+  end
+
+  def create
```
Comment (BregornOriginal) [inline app/controllers/api/comments_controller.rb]: - [ ] While creating a comment on the webpage I have the following error. Kindly fix this to be able to continue

image

File: config/routes.rb [hunk 2]
```diff
@@ -1,5 +1,7 @@
 
 Rails.application.routes.draw do
+  mount Rswag::Ui::Engine => '/api-docs'
+  mount Rswag::Api::Engine => '/api-docs'
```
Comment (BregornOriginal) [inline config/routes.rb:4]: - [ ] Kindly fix your routes to match the structure of your application. This means that to point to, for example `api/v1/users`, you should modify your structure in folders and add the corresponding routes to `config/routes`

**Review by BregornOriginal — CHANGES_REQUESTED** [review]
Hi @Team 

Good job so far!

There are some issues that you still need to work on to go to the next project but you are almost there!

cat-driving-serious

## Highlights

✅ Linters are passing
✅ Good commit history
✅ README looks professional


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by frank1738 — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Correct use of Github workflow
✅ No linter error

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**